src/zrc/sync/signals.py

```diff
@@ -90,10 +91,23 @@ def sync_delete(relation: ZaakInformatieObject):
         raise SyncError(f"Could not {operation} remote relation") from exc
 
 
-@receiver([post_save, post_delete], sender=ZaakInformatieObject, dispatch_uid='sync.sync_informatieobject_relation')
+@receiver([post_save, pre_delete], sender=ZaakInformatieObject, dispatch_uid='sync.sync_informatieobject_relation')
 def sync_informatieobject_relation(sender, instance: ZaakInformatieObject=None, **kwargs):
     signal = kwargs['signal']
     if signal is post_save and kwargs.get('created', False):
         sync_create(instance)
-    elif signal is post_delete:
+    elif signal is pre_delete:
+        # Add the uuid of the ZaakInformatieObject to the list of ZIOs that are
+        # marked for delete, causing them not to show up when performing
+        # GET requests on the ZRC, allowing the validation in the DRC to pass
+        marked_zios = cache.get('zios_marked_for_delete')
```

Comment:
De mogelijke race-conditie hier is wat mij betreft acceptabel maar ben benieuwd wat @sergei-maertens er van vind.

Reply:
![okay](https://tenor.com/ZHYH.gif)